### PR TITLE
fix: resolve `TransactionStream` and `Ledger` model issues

### LINKED ISSUE #2714: `TransactionStream` missing `close_time_iso` field
https://github.com/XRPLF/xrpl.js/blob/a0678857a16e4e11e37bf369b41e5c73a5c4f1a3/packages/xrpl/src/models/methods/subscribe.ts#L265-L297

https://xrpl.org/resources/dev-tools/websocket-api-tool#subscribe

with
```json
{
  "id": "Example watch one account and all new ledgers",
  "command": "subscribe",
  "streams": [
    "transactions"
  ]
}
```

```json
{
  "transaction": {
    "Account": "rpiFwLYi6Gb1ESHYorn2QG1WU5vw2u4exQ",
    "Fee": "10",
    "Flags": 0,
    "LastLedgerSequence": 88979521,
    "OfferSequence": 87142351,
    "Sequence": 0,
    "SigningPubKey": "ED4A8371720A1D404927F0B32DD49BDE4F8E2A8177D027482492AEE4F3C5045834",
    "TicketSequence": 87142355,
    "TransactionType": "OfferCancel",
    "TxnSignature": "6EBFB33ADCA039B8BCAAFA9080AB149C77334BB371F70B8DF06D038D8E1DA2867A52C690C0E2FE3E0A88EEEB101F3CD94AE20B4C471D979755759962B165E009",
    "hash": "1F04EA32F37227FD40E48151964EF46316C165E90E72E093E977F4DE594F0987",
    "date": 772811181
  },
  "meta": {
    "AffectedNodes": [
      {
        "ModifiedNode": {
          "FinalFields": {
            "Flags": 0,
            "IndexNext": "26a7",
            "IndexPrevious": "0",
            "Owner": "rpiFwLYi6Gb1ESHYorn2QG1WU5vw2u4exQ",
            "RootIndex": "F6ED7E4F8D3EAE3D246D421803B101F4C204AA91046C647121563B6F7DFD9310"
          },
          "LedgerEntryType": "DirectoryNode",
          "LedgerIndex": "2539B1BC3416189EDF08EC6FA3FD50CF7486840958FFB82B97F0D9C83268956E"
        }
      },
      {
        "ModifiedNode": {
          "FinalFields": {
            "Flags": 0,
            "IndexPrevious": "26ab",
            "Owner": "rpiFwLYi6Gb1ESHYorn2QG1WU5vw2u4exQ",
            "RootIndex": "F6ED7E4F8D3EAE3D246D421803B101F4C204AA91046C647121563B6F7DFD9310"
          },
          "LedgerEntryType": "DirectoryNode",
          "LedgerIndex": "57810DA107F2F9C57F254BBF1375C631F7F984BFF3E73AF2F07830C73966410F"
        }
      },
      {
        "DeletedNode": {
          "FinalFields": {
            "Account": "rpiFwLYi6Gb1ESHYorn2QG1WU5vw2u4exQ",
            "BookDirectory": "EC2927ADC0E519C170C4F169B7003F1A65A59C7410AD28865B06E26FE199FB10",
            "BookNode": "0",
            "Flags": 65536,
            "OwnerNode": "26ac",
            "PreviousTxnID": "BB08F10623795F1C9AC7605D955C537589FA495571ECE6DDFE91884B231B5AD0",
            "PreviousTxnLgrSeq": 88979486,
            "Sequence": 87142351,
            "TakerGets": {
              "currency": "EUR",
              "issuer": "raZdZ9eUZqHBo34mzZES39NP2PatznYP1d",
              "value": "186.515"
            },
            "TakerPays": "361432500"
          },
          "LedgerEntryType": "Offer",
          "LedgerIndex": "602FBA235986C31CB0872B161D5E0B9F2D9D40CEB576E32FA94665CC59F6C128"
        }
      },
      {
        "ModifiedNode": {
          "FinalFields": {
            "Account": "rpiFwLYi6Gb1ESHYorn2QG1WU5vw2u4exQ",
            "Balance": "8170048810",
            "Flags": 0,
            "OwnerCount": 154,
            "Sequence": 87142509,
            "TicketCount": 151
          },
          "LedgerEntryType": "AccountRoot",
          "LedgerIndex": "AB4F967F07A17B276A162D2BE831BB2B4BB7D28C58A71E2B47040114CE4599AE",
          "PreviousFields": {
            "Balance": "8170048820",
            "OwnerCount": 156,
            "TicketCount": 152
          },
          "PreviousTxnID": "1DD5C2F504141AB8B00F79DB9A1A091D98501389542C33CE016AFB45B377FE87",
          "PreviousTxnLgrSeq": 88979503
        }
      },
      {
        "DeletedNode": {
          "FinalFields": {
            "Account": "rpiFwLYi6Gb1ESHYorn2QG1WU5vw2u4exQ",
            "Flags": 0,
            "OwnerNode": "26a6",
            "PreviousTxnID": "0B6FE8AB7B37A88C494693E520AE4968112C44AA0B5EC20F9AB738BCDB8CBDEB",
            "PreviousTxnLgrSeq": 88979190,
            "TicketSequence": 87142355
          },
          "LedgerEntryType": "Ticket",
          "LedgerIndex": "CD7B539FF7EFB41AC031BA56659FEBC8018FB4A2A48A725F02DEE3ACA600A15E"
        }
      },
      {
        "DeletedNode": {
          "FinalFields": {
            "ExchangeRate": "5b06e26fe199fb10",
            "Flags": 0,
            "RootIndex": "EC2927ADC0E519C170C4F169B7003F1A65A59C7410AD28865B06E26FE199FB10",
            "TakerGetsCurrency": "0000000000000000000000004555520000000000",
            "TakerGetsIssuer": "3D04742494FF49328C57AD893BD2A02263416728",
            "TakerPaysCurrency": "0000000000000000000000000000000000000000",
            "TakerPaysIssuer": "0000000000000000000000000000000000000000"
          },
          "LedgerEntryType": "DirectoryNode",
          "LedgerIndex": "EC2927ADC0E519C170C4F169B7003F1A65A59C7410AD28865B06E26FE199FB10"
        }
      }
    ],
    "TransactionIndex": 26,
    "TransactionResult": "tesSUCCESS"
  },
  "type": "transaction",
  "validated": true,
  "status": "closed",
  "close_time_iso": "2024-06-27T13:46:21Z",
  "ledger_index": 88979503,
  "ledger_hash": "F730B80D09BE5982E7566BBAE918F9406A2554858DA3C88CB56E6E9A86FABC0A",
  "engine_result_code": 0,
  "engine_result": "tesSUCCESS",
  "engine_result_message": "The transaction was applied. Only final in a validated ledger."
}
```

### LINKED ISSUE #2715: Response of `ledger` command missing `close_time_iso` field
https://github.com/XRPLF/xrpl.js/blob/a0678857a16e4e11e37bf369b41e5c73a5c4f1a3/packages/xrpl/src/models/ledger/Ledger.ts#L11-L65

https://xrpl.org/resources/dev-tools/websocket-api-tool#ledger

```json
{
  "result": {
    "ledger_hash": "45008B805A9BCB7ABDD7D86C18936315AB55DA9DE0C4CB4C2210E45ECF244C81",
    "ledger_index": 88979737,
    "validated": true,
    "ledger": {
      "account_hash": "E50310BFE7B3BE4C03183167AC1307BD8CFAAAFCF13E3DE8D41A7614F7D6FF64",
      "close_flags": 0,
      "close_time": 772812072,
      "close_time_human": "2024-Jun-27 14:01:12.000000000 UTC",
      "close_time_resolution": 10,
      "close_time_iso": "2024-06-27T14:01:12Z",
      "ledger_hash": "45008B805A9BCB7ABDD7D86C18936315AB55DA9DE0C4CB4C2210E45ECF244C81",
      "parent_close_time": 772812071,
      "parent_hash": "3FCE0CC3ED85BD0CD9B00619D15E48AD8C725007D7030C5950F11B50E6139677",
      "total_coins": "99987462169597155",
      "transaction_hash": "A213F2F2C3D61BEB09AB39955FF3796B9F05033DD358F3076C9CC407A5A272DB",
      "ledger_index": "88979737",
      "closed": true
    }
  },
  "status": "success",
  "type": "response",
  "warnings": [
    {
      "id": 2001,
      "message": "This is a clio server. clio only serves validated data. If you want to talk to rippled, include 'ledger_index':'current' in your request"
    }
  ]
}
```

### LINKED ISSUE #2767: TransactionStream type is wrong when using API version 2
In [`src/models/methods/subscribe.ts`, line 289](https://github.com/XRPLF/xrpl.js/blob/f3960c3cccc26f7e595fc4365a68fdd52db03be3/packages/xrpl/src/models/methods/subscribe.ts#L289) the `TransactionStream` type declares a field called `transaction` which contains the main transaction data.

However, in reality, when using API version 2, this field appears to be called `tx_json` now.

It appears that similar changes were in fact taken into account in other places, such as [here in accountTx.ts](https://github.com/XRPLF/xrpl.js/blob/f3960c3cccc26f7e595fc4365a68fdd52db03be3/packages/xrpl/src/models/methods/accountTx.ts#L69-L75) but overlooked in this instance.

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

Title says it all.

### Context of Change

Fixes #2767
Fixes #2714
Fixes #2715

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

### Did you update HISTORY.md?

- [x] Yes

## Test Plan

The models match the output. CI passes.

### CODE REVIEW

File: packages/xrpl/src/client/partialPayment.ts [hunk 1]
```diff
@@ -159,10 +160,10 @@ export function handlePartialPayment<
  * @param log - The method used for logging by the connection (to report the partial payment).
  */
 export function handleStreamPartialPayment(
-  stream: TransactionStream,
+  stream: TransactionStream | TransactionV1Stream,
   log: (id: string, message: string) => void,
 ): void {
-  if (isPartialPayment(stream.transaction, stream.meta)) {
+  if (isPartialPayment(stream.transaction ?? stream.tx_json, stream.meta)) {
```
Comment (ckeshava) [inline packages/xrpl/src/client/partialPayment.ts:166]: I'd prefer it to be: 
```suggestion
  if (isPartialPayment(stream.tx_json ?? stream.transaction, stream.meta)) {
```

In the future rippled API versions, even if stream.transaction is deprecated/removed, this javascript code will still work correctly

**Review by ckeshava — APPROVED** [review]

**Comment by ckeshava** [comment]
The CodeRabbit AI does not make a mention of the changes pertaining to `tx_json` and `transaction` fields. 

I also don't understand the diagram because `TransactionStream` and `API` do not refer to anything concrete in the architecture of XRPL.

**Comment by mvadari** [comment]
> The CodeRabbit AI does not make a mention of the changes pertaining to `tx_json` and `transaction` fields.
> 
> I also don't understand the diagram because `TransactionStream` and `API` do not refer to anything concrete in the architecture of XRPL.

CodeRabbit isn't the smartest. cc @justinr1234

File: packages/xrpl/HISTORY.md [hunk 2]
```diff
@@ -7,6 +7,11 @@ Subscribe to [the **xrpl-announce** mailing list](https://groups.google.com/g/xr
 ### Added
 * parseTransactionFlags as a utility function in the xrpl package to streamline transactions flags-to-map conversion
 
+### Fixed
+* `TransactionStream` model supports APIv2
+* `TransactionStream` model includes `close_time_iso` field
+* `Ledger` model includes `close_time_iso` field
+
```
[the remark this answers is not in the document]
Comment (mvadari) [reply packages/xrpl/HISTORY.md:14]: This isn't version 1.2.5.

**Review by anissa-ripple — APPROVED** [review]

**Pull request merged**